src/main/java/com/example/onlinebookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,113 @@
+package com.example.onlinebookstore.service.impl;
+
+import com.example.onlinebookstore.dto.order.OrderDto;
+import com.example.onlinebookstore.dto.order.OrderRequestDto;
+import com.example.onlinebookstore.dto.order.OrderStatusDto;
+import com.example.onlinebookstore.dto.orderitem.OrderItemDto;
+import com.example.onlinebookstore.exception.EntityNotFoundException;
+import com.example.onlinebookstore.mapper.OrderItemMapper;
+import com.example.onlinebookstore.mapper.OrderMapper;
+import com.example.onlinebookstore.model.CartItem;
+import com.example.onlinebookstore.model.Order;
+import com.example.onlinebookstore.model.OrderItem;
+import com.example.onlinebookstore.model.ShoppingCart;
+import com.example.onlinebookstore.repository.CartItemRepository;
+import com.example.onlinebookstore.repository.OrderItemRepository;
+import com.example.onlinebookstore.repository.OrderRepository;
+import com.example.onlinebookstore.repository.ShoppingCartRepository;
+import com.example.onlinebookstore.service.OrderService;
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final CartItemRepository cartItemRepository;
+
+    @Transactional
+    @Override
+    public OrderDto createOrder(Long userId, OrderRequestDto requestDto) {
+        Order order = orderMapper.toEntity(requestDto);
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(userId);
+        if (shoppingCart == null) {
```

Comment:
for such options it's better to use Optional 

but could it be the case, when shop cart null for user? If you create shop cart during user registration, it's OneToOne mapping, I don't think we should check that cart is null
But this is not mistake